tests/Tests/ORM/Tools/SchemaToolTest.php

```diff
@@ -172,6 +172,23 @@ public function testPostGenerateEvents(): void
         self::assertTrue($listener->schemaCalled);
     }
 
+    public function testGetUpdateSchemaSqlEvents(): void
+    {
+        $listener = new GenerateSchemaEventListener();
+
+        $em = $this->getTestEntityManager();
+        $em->getEventManager()->addEventListener(
+            [ToolEvents::postGenerateSchema, ToolEvents::postGenerateComparisonSchema],
+            $listener,
+        );
+        $schemaTool = new SchemaTool($em);
+
+        $schema = $schemaTool->getUpdateSchemaSql([]);
+
+        self::assertTrue($listener->schemaCalled);
+        self::assertTrue($listener->comparisonSchemaCalled);
```

Comment:
The test does not really show me that I was able to influence the generated schema or SQL.

Reply:
Tried to improve the tests. This should show you that you can access the schema which the comparator uses.